opshin/type_inference.py

```diff
@@ -604,6 +605,7 @@ def visit_FunctionDef(self, node: FunctionDef) -> TypedFunctionDef:
         self.exit_scope()
         # We need the function type outside for usage
         self.set_variable_type(node.name, tfd.typ)
+        self.FUNCTION_ARGUMENT_REGISTRY[node.name] = node.args.args
```

Comment:
What happens for scoped overwrites? i.e. the following should be accepted
```
def foo(x:int):
    def foo(y:int):
        return y
    foo(y=1)
foo(x=1)
```

Reply:
`node.name` contains the unique name in the correct scope, with the original name being located in `node.orig_name`. In your example `foo(x:int)` would be called something like `'foo_0` and `foo(y:int)` would be named `foo_1`. In the `visit_Call` method `node.func.id` contains the name of the correct function taking scope into account.

Reply:
I see, neat trick. Please add a test case for this scenario and check the failing test cases in the CI.